ingestion/src/metadata/profiler/orm/functions/table_metric_computer.py

```diff
@@ -0,0 +1,463 @@
+        self._table = self._runner.table
+
+    @property
+    def database(self):
+        return self._database
+
+    @property
+    def table(self):
+        return self._table
+
+    @property
+    def runner(self):
+        return self._runner
+
+    @property
+    def metrics(self):
+        return self._metrics
+
+    @property
+    def table_name(self):
+        return self._table_name
+
+    @property
+    def schema_name(self):
+        return self._schema_name
+
+    @property
+    def conn_config(self):
+        return self._conn_config
+
+    def _set_table_and_schema_name(self):
+        """get table and schema name from table args
+
+        Args:
+            table (DeclarativeMeta): _description_
+        """
+        try:
+            self._schema_name = self.table.__table_args__.get("schema")
+            self._table_name = self.table.__tablename__
+        except AttributeError:
+            raise AttributeError(ERROR_MSG)
+
+    def _build_table(self, table, schema) -> Table:
+        """build table object from table name and schema name
+
+        Args:
+            table_name (str): table name
+            schema_name (str): schema name
+
+        Returns:
+            Table
+        """
+        if schema:
+            return Table(table, MetaData(), schema=schema)
+        return Table(table, MetaData())
+
+    def _get_col_names_and_count(self) -> Tuple[str, int]:
+        """get column names and count from table
+
+        Args:
+            table (DeclarativeMeta): table object
+
+        Returns:
+            Tuple[str, int]
+        """
+        col_names = literal(",".join(inspect(self.table).c.keys()), type_=String).label(
+            COLUMN_NAMES
+        )
+        col_count = literal(len(inspect(self.table).c)).label(COLUMN_COUNT)
+        return col_names, col_count
+
+    def _build_query(
+        self,
+        columns: List[Column],
+        table: Table,
+        where_clause: Optional[List[ColumnOperators]] = None,
+    ):
+
+        query = select(*columns).select_from(table)
+        if where_clause:
+            query = query.where(*where_clause)
+
+        return query
+
+    @abstractmethod
+    def compute(self):
+        """Default compute behavior for table metrics"""
+        raise NotImplementedError
+
+
+class BaseTableMetricComputer(AbstractTableMetricComputer):
+    """Base table computer"""
+
+    def _check_and_return(self, res):
+        """Check if the result is None and return the result or fallback
+
+        Args:
+            res (object): result
+        """
+        if res.rowCount is None:
+            return super().compute()
+        return res
+
+    def compute(self):
+        """Default compute behavior for table metrics"""
+        return self.runner.select_first_from_table(
+            *[metric().fn() for metric in self.metrics]
+        )
+
+
+class SnowflakeTableMetricComputer(BaseTableMetricComputer):
```

Comment:
Should we have a file structure similar to what we have for interfaces?
that would help keep the files small and specific. wdyt?

- orm
- - functions
- - - snowflake
- - - -table_metric_computer.py
- - - bigquery
- - - -table_metric_computer.py



something like that

Reply:
🤔 we could, though I am not sure it is necessary at this point. We have a small file ~400+ lines with tightly coupled class.

Reply:
okayy, maybe in the future when we add more dialects / dbs in the file, we can make it more readable in the directory structure. Approving Thanks 🙌